midend/flattenInterfaceStructs.h

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2018 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef _MIDEND_FLATTENINTERFACESTRUCT_H_
+#define _MIDEND_FLATTENINTERFACESTRUCT_H_
+
+#include "ir/ir.h"
+#include "frontends/p4/typeChecking/typeChecker.h"
+
+namespace P4 {
+
+/**
+   Describes how a type is replaced: the new type to
+   replace it and how each field is renamed.
+*/
+struct Replacement : public IHasDbPrint {
```

Comment:
You may want to find a better name for "Replacement". It is too generic and not nested in a specialized class.